nestjs Dockerfile failed using clean `create-nx-workspace@latest`

### Current Behavior

Create a new NX workspace with NestJS, but Docker fails to build and run.

### Expected Behavior

Should run the NestJS app in Docker - no code changes after setup,


### GitHub Repo

https://github.com/101sam/test-nest-docker

### Steps to Reproduce

```bash
npx create-nx-workspace@latest 

 NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

✔ Where would you like to create your workspace? · test-docker
✔ Which stack do you want to use? · node
✔ What framework should be used? · nest
✔ Application name · test-docker
✔ Would you like to generate a Dockerfile? [https://docs.docker.com/] · Yes
✔ Which unit test runner would you like to use? · jest
✔ Would you like to use ESLint? · Yes
✔ Would you like to use Prettier for code formatting? · Yes
✔ Which CI provider would you like to use? · github

 NX   Creating your v21.4.1 workspace.

✔ Installing dependencies with npm
✔ Successfully created the workspace: test-docker
✔ Nx Cloud has been set up successfully
✔ CI workflow has been generated successfully
✔ Would you like to push this workspace to Github? · Yes
✔ Repository name (format: username/repo-name): · 101sam/test-nest-docker
✓ Created repository 101sam/test-nest-docker on github.com
  https://github.com/101sam/test-nest-docker
✓ Added remote https://github.com/101sam/test-nest-docker.git
Enumerating objects: 53, done.
Counting objects: 100% (53/53), done.
Delta compression using up to 16 threads
Compressing objects: 100% (47/47), done.
Writing objects: 100% (53/53), 147.78 KiB | 16.42 MiB/s, done.
Total 53 (delta 3), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (3/3), done.
To https://github.com/101sam/test-nest-docker.git
 * [new branch]      HEAD -> main
branch 'main' set up to track 'origin/main'.
✓ Pushed commits to https://github.com/101sam/test-nest-docker.git

 NX   Successfully pushed to GitHub repository: https://github.com/101sam/test-nest-docker

 NX   Your CI setup is almost complete.
```

And then you try: 
1. `npx nx run @test-docker/test-docker:docker:build` 
2. `npx nx run @test-docker/test-docker:docker:run`


### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 22.17.1
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 11.5.2

nx                 : 21.4.1
@nx/js             : 21.4.1
@nx/jest           : 21.4.1
@nx/eslint         : 21.4.1
@nx/workspace      : 21.4.1
@nx/devkit         : 21.4.1
@nx/eslint-plugin  : 21.4.1
@nx/nest           : 21.4.1
@nx/node           : 21.4.1
@nx/web            : 21.4.1
@nx/webpack        : 21.4.1
@nx/docker         : 21.4.1
typescript         : 5.8.3
---------------------------------------
Registered Plugins:
@nx/js/typescript
@nx/webpack/plugin
@nx/eslint/plugin
@nx/jest/plugin
@nx/docker
---------------------------------------
Cache Usage: 293.01 KB / 185.82 GB
```

### Failure Logs

```shell
docker build . --tag apps-test-docker

[+] Building 0.3s (9/9) FINISHED                                                             docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                         0.0s
 => => transferring dockerfile: 594B                                                                         0.0s
 => [internal] load metadata for docker.io/library/node:lts-alpine                                           0.2s
 => [internal] load .dockerignore                                                                            0.0s
 => => transferring context: 2B                                                                              0.0s
 => [1/4] FROM docker.io/library/node:lts-alpine@sha256:1b2479dd35a99687d6638f5976fd235e26c5b37e8122f786fcd  0.0s
 => [internal] load build context                                                                            0.0s
 => => transferring context: 247B                                                                            0.0s
 => CACHED [2/4] WORKDIR /app                                                                                0.0s
 => CACHED [3/4] COPY dist .                                                                                 0.0s
 => CACHED [4/4] RUN npm --omit=dev -f install                                                               0.0s
 => exporting to image                                                                                       0.0s
 => => exporting layers                                                                                      0.0s
 => => writing image sha256:e3b837ca2f790c281592fdf9029ae4fcacd37bd582fb1a7af70862dc57f23933                 0.0s
 => => naming to docker.io/library/apps-test-docker                                                          0.0s

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/anef0o6wrpfxt86b66v9sf6i8

What's next:
    View a summary of image vulnerabilities and recommendations → docker scout quickview 

> nx run @test-docker/test-docker:docker:run

> docker run   apps-test-docker

node:internal/modules/cjs/loader:1368
  throw err;
  ^

Error: Cannot find module '@nestjs/common'
Require stack:
- /app/main.js
    at Function._resolveFilename (node:internal/modules/cjs/loader:1365:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1021:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1026:22)
    at Function._load (node:internal/modules/cjs/loader:1175:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at Module.require (node:internal/modules/cjs/loader:1445:12)
    at require (node:internal/modules/helpers:135:16)
    at Array.<anonymous> (/app/main.js:8:18)
    at __webpack_require__ (/app/main.js:118:41) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/app/main.js' ]
}

Node.js v22.18.0
```

### Package Manager Version

_No response_

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/nest/src/generators/application/application.spec.ts
+++ b/packages/nest/src/generators/application/application.spec.ts
@@ -260,7 +260,17 @@ describe('application generator', () => {
       `);
       expect(readJson(tree, 'myapp/package.json')).toMatchInlineSnapshot(`
         {
-          "dependencies": {},
+          "dependencies": {
+            "@nestjs/common": "^11.0.0",
+            "@nestjs/core": "^11.0.0",
+            "@nestjs/platform-express": "^11.0.0",
+            "reflect-metadata": "^0.1.13",
+            "rxjs": "^7.8.0",
+            "tslib": "^2.3.0",
+          },
+          "devDependencies": {
+            "@nestjs/testing": "^11.0.0",
+          },
           "name": "@proj/myapp",
           "nx": {
             "targets": {
@@ -442,6 +452,7 @@ describe('application generator', () => {
           "private",
           "nx",
           "dependencies",
+          "devDependencies",
         ]
       `);
     });

--- a/packages/nest/src/generators/application/application.ts
+++ b/packages/nest/src/generators/application/application.ts
@@ -44,7 +44,13 @@ export async function applicationGeneratorInternal(
   updateTsConfig(tree, options);
 
   if (!options.skipPackageJson) {
+    // Install dependencies to root package.json
     tasks.push(ensureDependencies(tree));
+
+    // Install dependencies to project's package.json (for PM Workspaces)
+    if (tree.exists(`${options.appProjectRoot}/package.json`)) {
+      tasks.push(ensureDependencies(tree, options.appProjectRoot));
+    }
   }
 
   if (!options.skipFormat) {

--- a/packages/nest/src/utils/ensure-dependencies.ts
+++ b/packages/nest/src/utils/ensure-dependencies.ts
@@ -1,5 +1,5 @@
 import type { GeneratorCallback, Tree } from '@nx/devkit';
-import { addDependenciesToPackageJson } from '@nx/devkit';
+import { addDependenciesToPackageJson, joinPathFragments } from '@nx/devkit';
 import {
   nestJsVersion,
   reflectMetadataVersion,
@@ -7,7 +7,14 @@ import {
   tsLibVersion,
 } from './versions';
 
-export function ensureDependencies(tree: Tree): GeneratorCallback {
+export function ensureDependencies(
+  tree: Tree,
+  projectRoot?: string
+): GeneratorCallback {
+  const packageJsonPath = projectRoot
+    ? joinPathFragments(projectRoot, 'package.json')
+    : 'package.json';
+
   return addDependenciesToPackageJson(
     tree,
     {
@@ -20,6 +27,7 @@ export function ensureDependencies(tree: Tree): GeneratorCallback {
     },
     {
       '@nestjs/testing': nestJsVersion,
-    }
+    },
+    packageJsonPath
   );
 }

